### PR TITLE
Git lab add trigger pipeline command

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/CIAC-5322

## Description
Added a command to trigger a pipeline (nightly) on GitLab.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.py [hunk 1]
```diff
@@ -314,6 +315,10 @@ def get_job_artifact_request(self, project_id: str, job_id: str, artifact_path_s
         suffix = f'projects/{project_id}/jobs/{job_id}/artifacts/{artifact_path_suffix}'
         return self._http_request('get', suffix, headers=headers, resp_type='text')
 
+    def gitlab_trigger_pipeline(self, project_id: str, data: dict):
```
Comment (bziser) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.py:318]: please add return type and description to the function

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.py [hunk 2]
```diff
@@ -1698,6 +1703,42 @@ def gitlab_artifact_get_command(client: Client, args: Dict[str, Any]) -> Command
...
 
 
+def gitlab_trigger_pipeline_command(client: Client, args: dict[str, str]) -> CommandResults:
+    """
+    Triggers a GitLab pipeline on a selected project and branch.
+    Args:
+        client (Client): Client to perform calls to GitLab services.
+        args (dict) XSOAR arguments:
+            - 'project_id': Project ID on winch to run the pipeline.
+            - 'ref_branch': The branch on which to run the pipeline. Default is 'master'
```
Comment (bziser) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.py:1713]:
```suggestion
            - 'project_id': Project ID on which to run the pipeline.
            - 'ref_branch': The branch on which to run the pipeline. Default is 'master'
```

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.py [hunk 3]
```diff
@@ -1698,6 +1703,42 @@ def gitlab_artifact_get_command(client: Client, args: Dict[str, Any]) -> Command
...
+
+    Returns:
+        (CommandResults).
+    """
+    project_id = args.get('project_id') or client.project_id
```
Comment (bziser) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.py:1718]: can it be that we will get `project_id=None`? what will happen in such case?

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2_test.py [hunk 4]
```diff
@@ -986,3 +986,33 @@ def test_return_date_arg_as_iso(arg, isValidDate, expected_response):
             return_date_arg_as_iso(arg)
 
         assert str(e.value) == expected_response
+
+
+@pytest.mark.parametrize('trigger_token, args, expected_result', [
+    ('', {}, util_load_json('test_data/commands_test_data.json').get('trigger_pipeline1')),
+    (1111, {'project_id': 2222, 'ref_branch': 'test'}, util_load_json('test_data/commands_test_data.json').get('trigger_pipeline2'))
+])
+def test_trigger_pipeline(mocker, trigger_token, args, expected_result):
```
Comment (bziser) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2_test.py:995]: please ass `given,when,then`

**Review by bziser — COMMENTED** [review]
looks good

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.yml [hunk 5]
```diff
@@ -13,6 +13,12 @@ configuration:
   hiddenusername: true
   required: true
   type: 9
+- displaypassword: Trigger Token
+  additionalinfo: The trigger token in order to run pipelines. For more info see https://docs.gitlab.com/ee/ci/triggers
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.yml:17]:
```suggestion
  additionalinfo: The trigger token to run pipelines. For more info see https://docs.gitlab.com/ee/ci/triggers
```

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.yml [hunk 6]
```diff
@@ -3664,6 +3670,80 @@ script:
     - contextPath: GitLab.Artifact.artifact_data
       description: Data of the artifact requested.
       type: String
+  - name: gitlab-trigger-pipeline
+    arguments:
+    - default: false
+      description: Project ID on winch to run the pipeline.
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.yml:3676]:
```suggestion
      description: Project ID on which to run the pipeline.
```

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.yml [hunk 7]
```diff
@@ -3664,6 +3670,80 @@ script:
...
+      required: false
+      secret: false
+    - default: false
+      description: The branch on which to run the pipeline. Default is 'master'.
+      isArray: false
+      name: ref_branch
+      required: false
+      secret: false
+    - default: false
+      description: json contains the pipeline variables.
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.yml:3688]:
```suggestion
      description: JSON containing the pipeline variables.
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 8]
```diff
@@ -2144,3 +2144,42 @@ Gets an artifact from a given artifact path, corresponding to a given job ID.
...
+
+#### Base Command
+
+`gitlab-trigger-pipeline`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| project_id | Project ID on winch to run the pipeline. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:2161]:
```suggestion
| project_id | Project ID on which to run the pipeline. | Optional | 
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 9]
```diff
@@ -2144,3 +2144,42 @@ Gets an artifact from a given artifact path, corresponding to a given job ID.
...
+
+`gitlab-trigger-pipeline`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| project_id | Project ID on winch to run the pipeline. | Optional | 
+| ref_branch | The branch on which to run the pipeline. Default is 'master'. | Optional | 
+| trigger_variables | json contains the pipeline variables. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:2163]:
```suggestion
| trigger_variables | JSON containing the pipeline variables. | Optional | 
```

File: Packs/GitLab/ReleaseNotes/2_2_11.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### GitLab v2
+- Added a new command **gitlab-trigger-pipeline** to trigger a GitLab pipeline
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/ReleaseNotes/2_2_11.md:4]:
```suggestion
Added the new ***gitlab-trigger-pipeline*** command to trigger a GitLab pipeline.
```

**Comment by ShirleyDenkberg** [comment]
@bziser @DeanArbel Doc review completed.

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2.py [hunk 11]
```diff
@@ -314,6 +315,19 @@ def get_job_artifact_request(self, project_id: str, job_id: str, artifact_path_s
         suffix = f'projects/{project_id}/jobs/{job_id}/artifacts/{artifact_path_suffix}'
         return self._http_request('get', suffix, headers=headers, resp_type='text')
 
+    def gitlab_trigger_pipeline(self, project_id: str, data: dict) -> dict:
+        """Triggers a pipeline on GitLab.
+
+        Args:
+            project_id: Project ID on winch to run the pipeline.
```
Comment (bziser) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2.py:322]:
```suggestion
            project_id: Project ID on which to run the pipeline.
```

File: Packs/GitLab/Integrations/GitLabv2/GitLabv2_test.py [hunk 12]
```diff
@@ -986,3 +986,45 @@ def test_return_date_arg_as_iso(arg, isValidDate, expected_response):
...
+            - case 2 - The response is correct with the same branch and project_id as in the args.
+    """
+    from GitLabv2 import Client, gitlab_trigger_pipeline_command
+    client = Client(project_id=1234,
+                    base_url="base_url",
+                    verify=False,
+                    proxy=False,
+                    headers={'PRIVATE-TOKEN': 'api_key'},
+                    trigger_token=trigger_token)
+    expected_outputs: List[Dict] = expected_result['expected_outputs']
```
Comment (bziser) [inline Packs/GitLab/Integrations/GitLabv2/GitLabv2_test.py:1015]:
```suggestion
    expected_outputs: list[dict] = expected_result['expected_outputs']
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 13]
```diff
@@ -28,6 +29,18 @@ To generate a new token:
 6. Select the **scopes**, or **permissions**, you want to grant this token. The minimum is read-only on repo.
 7. Click **Create personal access token** and copy the api key generated.-+
 
+#### Create a Trigger Token
+Trigger tokens allows you to trigger a pipeline for a branch using it to authenticate on an API call.
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:33]:
```suggestion
Trigger tokens allow you to trigger a pipeline for a branch using it to authenticate on an API call.
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 14]
```diff
@@ -28,6 +29,18 @@ To generate a new token:
...
+Prerequisite:
+- You must have at least the Maintainer role for the project.
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:35]:
```suggestion
You must have at least the Maintainer role for the project.
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 15]
```diff
@@ -28,6 +29,18 @@ To generate a new token:
...
+To generate a new token:
+1. Navigate to your project.
+2. Select Settings > CI/CD.
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:38]:
```suggestion
2. Select **Settings** > **CI/CD**.
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 16]
```diff
@@ -28,6 +29,18 @@ To generate a new token:
...
+3. Expand Pipeline triggers.
+4. Enter a description and select Add trigger.
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:40]:
```suggestion
4. Enter a description and select **Add trigger**.
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 17]
```diff
@@ -28,6 +29,18 @@ To generate a new token:
 6. Select the **scopes**, or **permissions**, you want to grant this token. The minimum is read-only on repo.
 7. Click **Create personal access token** and copy the api key generated.-+
 
+#### Create a Trigger Token
+Trigger tokens allows you to trigger a pipeline for a branch using it to authenticate on an API call.
+Prerequisite:
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:34]:
```suggestion
 
 **Prerequisite:**
 
```

File: Packs/GitLab/Integrations/GitLabv2/README.md [hunk 18]
```diff
@@ -28,6 +29,18 @@ To generate a new token:
...
+- You must have at least the Maintainer role for the project.
+To generate a new token:
```
Comment (ShirleyDenkberg) [inline Packs/GitLab/Integrations/GitLabv2/README.md:36]:
```suggestion

**To generate a new token:**
```

**Comment by ShirleyDenkberg** [comment]
@MosheEichler Reviewed doc updates as requested.

**Review by bziser — APPROVED** [review]

**Review by bziser — APPROVED** [review]

**Pull request merged**